File: src/main/java/com/langleon/dsobuildsim/wisdomskilltree/dto/definition/WisdomSkillDefinitionDTO.java
package com.langleon.dsobuildsim.wisdomskilltree.dto.definition;

import com.langleon.dsobuildsim.common.StatType;
import com.langleon.dsobuildsim.wisdomskilltree.wisdomskill.WisdomSkillType;

public record WisdomSkillDefinitionDTO(
        WisdomSkillType type,
        String name,
        int maxLevel,
        int currentLevel,
        int costPerLevel,
        StatType statType,
        double increasePerLevel,
        double firstLevelBonus,
        String description
) {
}

File: frontend/src/app/models/gamedataDTOs/WisdomSkillDefinitionDTO.ts
import { StatType } from "../../enums/StatType";
import { WisdomSkillType } from "../../enums/WisdomSkillType";

export interface WisdomSkillDefinitionDTO {
    type: WisdomSkillType;
    name: string;
    maxLevel: number;
    currentLevel: number;   
    costPerLevel: number;
    statType: StatType;
    increasePerLevel: number;
    firstLevelBonus: number;
    description: string;
}